src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,373 @@
+            cancellationToken As CancellationToken) As MultiLineIfBlockSyntax
+
+            Dim ifPart = ifNode
+            Dim elseBlock = ifNode.ElseBlock
+
+            Dim ifStatement = ifNode.IfStatement
+
+            Dim ifLeadingTrivia = ifNode.GetLeadingTrivia()
+            Dim endifTrailingTrivia = ifNode.EndIfStatement.GetTrailingTrivia()
+            Dim elseBlockLeadingTrivia = elseBlock.GetLeadingTrivia()
+            Dim endifLeadingTrivia = ifNode.EndIfStatement.GetLeadingTrivia()
+
+            Return ifNode _
+                .WithIfStatement(ifStatement.WithCondition(Negate(ifStatement.Condition, semanticModel, cancellationToken))) _
+                .WithStatements(elseBlock.Statements) _
+                .WithElseBlock(elseBlock.WithStatements(ifPart.Statements).WithLeadingTrivia(endifLeadingTrivia)) _
+                .WithEndIfStatement(ifNode.EndIfStatement.WithTrailingTrivia(endifTrailingTrivia).WithLeadingTrivia(elseBlockLeadingTrivia)) _
+                .WithLeadingTrivia(ifLeadingTrivia)
+        End Function
+
+        Private Function TryNegateBinaryComparisonExpression(
+            expression As ExpressionSyntax,
+            semanticModel As SemanticModel,
+            cancellationToken As CancellationToken,
+            ByRef result As ExpressionSyntax) As Boolean
+
+            Dim inverses As Tuple(Of SyntaxKind, SyntaxKind) = Nothing
+            If s_comparisonInversesMap.TryGetValue(expression.Kind, inverses) Then
+                Dim binaryExpression = DirectCast(expression, BinaryExpressionSyntax)
+                Dim expressionType = inverses.Item1
+                Dim operatorType = inverses.Item2
+
+                ' Special case negating Length > 0 to Length = 0 and 0 < Length to 0 == Length
+                ' for arrays and strings. We can do this because we know that Length cannot be
+                ' less than 0.
+                Dim operation = semanticModel.GetOperation(binaryExpression)
+
+                If (IsSpecialCaseBinaryExpression(TryCast(operation, IBinaryOperation), cancellationToken)) Then
+                    expressionType = SyntaxKind.EqualsExpression
+                    operatorType = SyntaxKind.EqualsToken
+                End If
+
+                result = SyntaxFactory.BinaryExpression(
+                    expressionType,
+                    binaryExpression.Left.Parenthesize(),
+                    SyntaxFactory.Token(
+                        binaryExpression.OperatorToken.LeadingTrivia,
+                        operatorType,
+                        binaryExpression.OperatorToken.TrailingTrivia),
+                    binaryExpression.Right.Parenthesize())
+
+                result = result _
+                .WithLeadingTrivia(binaryExpression.GetLeadingTrivia()) _
+                .WithTrailingTrivia(binaryExpression.GetTrailingTrivia())
+
+                Return True
+            End If
+
+            Return False
+        End Function
```

Comment:
this logic looks nearly identical to the C# side.  Why can't we share this code in the base class?